### ISSUE TITLE
[bug] conan inspect does not default to looking for python_requires in remotes

### ISSUE THREAD

**Issue by geir-t** [conan-io/conan#14571]
### Environment details

- Operating System+version: Ubuntu 22.04
- Compiler+version: N/A
- Conan version: 2.0.9
- Python version: 3.10.8

### Steps to reproduce

Packages with python_requires in conan2 does not automatically look in remotes, even though this is default behavior for other commands like conan create
```
conan inspect .
ERROR: Error loading conanfile at '/home/user/repos/package/conanfile.py': Version range '>=0.0.1,<1.0' from requirement 'python_requires_package/[>=0.0.1,<1.0]@company, Traits: build=False, headers=True, libs=True, run=False, visible=True' required by 'py_require' could not be resolved
```

For conan1 the behaviour is opposite, it looks in remotes
```
conan inspect . -a version
python_requires_package/0.1.0+39.sha.1c47355@company: Not found in local cache, looking in remotes...
python_requires_package/0.1.0+39.sha.1c47355@company: Trying with 'conan'...
Downloading conanmanifest.txt completed [0.06k]                                          
Downloading conanfile.py completed [3.05k]                                               
python_requires_package/0.1.0+39.sha.1c47355@company: Downloaded recipe revision c525c55123b4d0912b0fbc5e2f1f9b10
version: 0.0.2+dirty.0.sha.84accf8
```

**Comment by AbrilRBS** [comment]
Hi! Thanks a lot for your report

This was reported to us also yesterday, so the PR was almost ready :)

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/14572 merged, this will be in next 2.0.10

**Comment by geir-t** [comment]
> Hi! Thanks a lot for your report
> 
> This was reported to us also yesterday, so the PR was almost ready :)

That was fast, thanks :)
